sdcm/cluster.py

```diff
@@ -302,41 +301,57 @@ def init(self) -> None:
                              exc_info=True)
         self._init_argus_resource()
 
+    @cached_property
+    def argus_client(self) -> ArgusSCTClient | None:
+        try:
+            run_id = self.test_config.test_id()
+            creds = KeyStore().get_argus_rest_credentials()
+            client = ArgusSCTClient(run_id=run_id, auth_token=creds["token"], base_url=creds["baseUrl"])
+            return client
+        except Exception:  # pylint: disable=broad-except
+            LOGGER.error("Error creating argus client", exc_info=True)
+        return None
+
     def _init_argus_resource(self):
         try:
-            run = ArgusTestRun.get()
+
+            if not (client := self.argus_client):
+                return
             shards = -1 if "db-node" in self.instance_name else self.cpu_cores
-            instance_details = CloudInstanceDetails(public_ip=self.public_ip_address, region=self.region,
-                                                    provider=self.parent_cluster.cluster_backend,
-                                                    private_ip=self.ip_address, creation_time=int(time.time()),
-                                                    shards_amount=shards)
-            resource = CloudResource(name=self.name, state=ResourceState.RUNNING,
-                                     instance_info=instance_details, resource_type=self.node_type)
-            self.argus_resource = resource
-            run.run_info.resources.attach_resource(resource)
-            run.save()
+            self.argus_resource = True
+            client.create_resource(
+                name=self.name,
+                resource_type=self.node_type,
+                public_ip=self.public_ip_address,
+                private_ip=self.ip_address,
+                region=self.region,
+                provider=self.parent_cluster.cluster_backend,
+                shards_amount=shards,
+                state=ResourceState.RUNNING,
+            )
         except Exception:  # pylint: disable=broad-except
             LOGGER.error("Encountered an unhandled exception while interacting with Argus", exc_info=True)
 
-    def argus_resource_set_shards(self) -> CloudResource | None:
+    def argus_resource_set_shards(self) -> bool:
```

Comment:
why this need to return boolean, the code doesn't check it.

Reply:
also rename it to `argus_set_shards`, `argus_resource` doesn't exists anymore

Reply:
Done